src/test/java/uk/gov/homeoffice/digital/sas/timecard/producers/KafkaProducerTimeEntryTest.java

```diff
@@ -0,0 +1,89 @@
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Date;
+import java.util.UUID;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.SendResult;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.util.concurrent.ListenableFuture;
+import uk.gov.homeoffice.digital.sas.timecard.enums.KafkaAction;
+import uk.gov.homeoffice.digital.sas.timecard.model.KafkaEventMessage;
+import uk.gov.homeoffice.digital.sas.timecard.model.TimeEntry;
+
+@ExtendWith(SpringExtension.class)
+class KafkaProducerTimeEntryTest {
+
+  @Mock
+  private KafkaTemplate<String, KafkaEventMessage> kafkaTimeEntryTemplate;
+
+  @InjectMocks
+  private KafkaProducerTimeEntry kafkaProducerTimeEntry;
+
+  @ParameterizedTest
+  @EnumSource(value = KafkaAction.class, names = {"CREATE", "UPDATE"})
+  void sendMessage_actionOnTimeEntry_messageIsSentWithCorrectArguments(KafkaAction action) {
+    TimeEntry timeEntry = createTimeEntry();
+
+    ListenableFuture<SendResult<String, KafkaEventMessage>> responseFuture =
+        mock(ListenableFuture.class);
+    Mockito.when(kafkaTimeEntryTemplate.send(Mockito.any(), Mockito.any(), Mockito.any()))
+        .thenReturn(responseFuture);
+
+    kafkaProducerTimeEntry.sendMessage(timeEntry, action);
+
+    ArgumentCaptor<String> topicArgument = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<String> ownerIdArgument = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<KafkaEventMessage> messageArgument =
+        ArgumentCaptor.forClass(KafkaEventMessage.class);
+
+    Mockito.verify(kafkaTimeEntryTemplate)
+        .send(topicArgument.capture(), ownerIdArgument.capture(), messageArgument.capture());
+
+    assertEquals("callisto-timecard", topicArgument.getValue());
+    assertEquals(timeEntry.getOwnerId().toString(), ownerIdArgument.getValue());
+    assertEquals(timeEntry, messageArgument.getValue().getResource());
+    assertEquals(action, messageArgument.getValue().getAction());
+  }
+
+  @ParameterizedTest
+  @EnumSource(value = KafkaAction.class, names = {"CREATE", "UPDATE"})
+  void sendMessage_actionOnTimeEntry_noExceptionThrown(KafkaAction action) {
+    TimeEntry timeEntry = createTimeEntry();
+
+    ListenableFuture<SendResult<String, KafkaEventMessage>> responseFuture =
+        mock(ListenableFuture.class);
+    Mockito.when(kafkaTimeEntryTemplate.send(Mockito.any(), Mockito.any(), Mockito.any()))
+        .thenReturn(responseFuture);
+
+    assertThatNoException().isThrownBy(() ->
+        kafkaProducerTimeEntry.sendMessage(timeEntry, action));
+  }
+
+  private TimeEntry createTimeEntry() {
```

Comment:
I think now that we've got two test classes depending on the time entry entity (this & `TimeEntryValidatorTest`) it would be a good time to have a test util class that creates time entries for us. 

Perhaps a final class with a private constructor and static methods to create different types of time entries. Might save us trouble down the line as more tests use it.

Reply:
Moved these functions into a `TimeEntryFactory` class and the `getAsDate` function a `CommonUtils` class

Reply:
Looks good, just one minor comment about the utils package name. 
'Utils' (under the test package) sounds like its testing util classes from the main app.
Could we rename the pkg to something like 'TestUtils' to imply they're only to assist the tests?

Reply:
Updated now